### PR TITLE
[ci] Add amazonlinux 2 and 2023 to platform-support pipeline

### PR DESCRIPTION
**Pull request by brianseeders**

### CODE REVIEW

File: .buildkite/pipelines/periodic-platform-support.yml [hunk 1]
```diff
@@ -1,71 +1,46 @@
...
-        agents:
-          provider: gcp
-          image: family/elasticsearch-{{matrix.image}}
-          machineType: n1-standard-32
-          diskType: pd-ssd
-          diskSizeGb: 350
-        env:
-          GRADLE_TASK: "{{matrix.GRADLE_TASK}}"
-  - group: platform-support-arm
+  # - group: platform-support-unix
```
Comment (mark-vieira) [inline .buildkite/pipelines/periodic-platform-support.yml:2]: Presumably this stuff is commented out just to facilitate more efficient testing?

Comment (brianseeders) [reply .buildkite/pipelines/periodic-platform-support.yml:2]: This is a draft PR, get out of here

Comment (mark-vieira) [reply .buildkite/pipelines/periodic-platform-support.yml:2]: Then don't request a review yet 😉 
image

Comment (brianseeders) [reply .buildkite/pipelines/periodic-platform-support.yml:2]: Did you get a notification or something? I thought it didn't really get pushed out until the PR went to Ready. Maybe that's only with `CODEOWNERS`.

Comment (mark-vieira) [reply .buildkite/pipelines/periodic-platform-support.yml:2]: I received an email, yes.

**Comment by brianseeders** [comment]
test this

**Comment by brianseeders** [comment]
buildkite test this please

**Pull request closed without merging**